packages/web3-core/src/web3_subscriptions.ts

```diff
@@ -77,13 +77,16 @@ export abstract class Web3Subscription<
 	}
 
 	public async unsubscribe() {
-		await this._requestManager.send<string>({
+		if (!this.id) {
```

Comment:
Its not optional `private _id?: HexString;`

so you can adjust this in id def and in this function as well.

Reply:
It's not optional but its initialized after you subscribed and set to `undefined` when you `unsubscribe`. So this check make sure you don't unsubscribe a non-subscribed or already unsubscribe subscription.